controllers/datafoundationclient.go

```diff
@@ -0,0 +1,217 @@
+package controllers
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	opv1a1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	v1alpha1 "github.com/red-hat-storage/managed-fusion-agent/api/v1alpha1"
+	ocsclient "github.com/red-hat-storage/ocs-client-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	ocsClientOperatorName = "ocs-client-operator"
+)
```

Comment:
Please move consts and vars section below type definitions.

Reply:
updated